.github/workflows/test.yaml

```diff
@@ -0,0 +1,37 @@
+name: Test
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+env:
+  GOPRIVATE: "github.com/speakeasy-api"
+jobs:
+  build:
+    if: github.event.pull_request.draft == false
```

Comment:
would you really not want to run these on drafts? I think one of the main benefits of a draft is being able to run them through the github actions

Reply:
No strong preferences.

I generally block github actions running on `drafts` as a matter of habit; with the assumption that the PR usually exists for some form of feedback gathering, but are often in a non-working state. Since tests often take a while to execute, the habit is just a little bit of frugality.

In the case of `go test`, since it executes in seconds, there's nothing really lost by executing it always. If you've a strong opinion here happy to adhere to it.